service/src/test/java/bio/terra/pipelines/dependencies/cbas/CbasServiceTest.java

```diff
@@ -0,0 +1,183 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.doReturn;
+
+import bio.terra.cbas.api.MethodsApi;
+import bio.terra.cbas.api.PublicApi;
+import bio.terra.cbas.client.ApiException;
+import bio.terra.cbas.model.MethodDetails;
+import bio.terra.cbas.model.MethodListResponse;
+import bio.terra.cbas.model.SystemStatus;
+import bio.terra.pipelines.app.configuration.external.CbasConfiguration;
+import bio.terra.pipelines.app.configuration.internal.RetryConfiguration;
+import bio.terra.pipelines.dependencies.common.HealthCheckWorkspaceApps;
+import java.net.SocketTimeoutException;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.support.RetryTemplate;
+
+@ExtendWith(MockitoExtension.class)
+class CbasServiceTest {
+  final RetryConfiguration retryConfig = new RetryConfiguration();
+  RetryTemplate template = retryConfig.listenerResetRetryTemplate();
+
+  final CbasConfiguration cbasConfiguration = new CbasConfiguration(false);
+
+  final Answer<Object> errorAnswer =
+      invocation -> {
+        throw new SocketTimeoutException("Timeout");
+      };
+
+  private final String cbaseBaseUri = "baseUri";
+  private final String accessToken = "accessToken";
+
+  @BeforeEach
+  void init() {
+    FixedBackOffPolicy smallerBackoff = new FixedBackOffPolicy();
+    smallerBackoff.setBackOffPeriod(5L); // 5 ms
+    template.setBackOffPolicy(smallerBackoff);
+  }
+
+  @Test
+  void socketExceptionRetriesEventuallySucceed() throws Exception {
+    MethodListResponse expectedResponse =
+        new MethodListResponse().addMethodsItem(new MethodDetails().name("methodName"));
+
+    CbasClient cbasClient = mock(CbasClient.class);
+    MethodsApi methodsApi = mock(MethodsApi.class);
+    when(methodsApi.getMethods(any(), any(), any()))
+        .thenAnswer(errorAnswer)
+        .thenReturn(expectedResponse);
+
+    CbasService cbasService = spy(new CbasService(cbasClient, template));
+
+    doReturn(methodsApi).when(cbasClient).methodsApi(any(), any());
+
+    assertEquals(expectedResponse, cbasService.getAllMethods(cbaseBaseUri, accessToken));
+  }
+
+  @Test
+  void socketExceptionRetriesEventuallyFail() throws Exception {
+    MethodListResponse expectedResponse =
+        new MethodListResponse().addMethodsItem(new MethodDetails().name("methodName"));
+
+    CbasClient cbasClient = mock(CbasClient.class);
+    MethodsApi methodsApi = mock(MethodsApi.class);
+    when(methodsApi.getMethods(any(), any(), any()))
+        .thenAnswer(errorAnswer)
+        .thenAnswer(errorAnswer)
+        .thenAnswer(errorAnswer)
+        .thenReturn(expectedResponse);
```

Comment:
minor: maybe add a comment somewhere that the max number of attempts is defined as 3 in `RetryConfiguration` (this is the kind of thing that now that i've looked this up i'll never have this question again, but coming to this fresh it took me a few minutes to track that down. so not a huge deal, just would be helpful.)

Reply:
added!